modules/apps/frontend-view-state/frontend-view-state-impl/src/main/java/com/liferay/frontend/view/state/internal/active/FVSActiveSettingsFactoryImpl.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.view.state.internal.active;
+
+import com.liferay.frontend.view.state.active.FVSActiveSettings;
+import com.liferay.frontend.view.state.active.FVSActiveSettingsFactory;
+import com.liferay.frontend.view.state.model.FVSActiveEntry;
+import com.liferay.frontend.view.state.model.FVSEntry;
+import com.liferay.frontend.view.state.service.FVSActiveEntryLocalService;
+import com.liferay.frontend.view.state.service.FVSEntryLocalService;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = FVSActiveSettingsFactory.class)
+public class FVSActiveSettingsFactoryImpl implements FVSActiveSettingsFactory {
+
+	public FVSActiveSettings getFVSActiveSettings(
+		HttpServletRequest httpServletRequest, String clayDatasetDisplayId) {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
+
+		FVSActiveEntry fvsActiveEntry =
+			_fvsActiveEntryLocalService.fetchFVSActiveEntry(
+				themeDisplay.getUserId(), clayDatasetDisplayId,
+				themeDisplay.getPlid(), portletDisplay.getId());
+
+		FVSEntry fvsEntry;
+
+		if (fvsActiveEntry == null) {
+			fvsEntry = _fvsEntryLocalService.createFVSEntry("{}");
```

Comment:
Do you need to create an entry when it does not exists? when is this going to be called? when the page displays the dataset or when the users wants to create a customization?

Reply:
This will be used by the migration of portlet preferences in `frontend-taglib-clay`. That's the [new LPS](https://issues.liferay.com/browse/LPS-132515) that we cannot implement now because of the near release

Reply:
Note that active view settings is not the same as custom view settings (which we have not implemented yet). Active view settings are the equivalent of the current configuration stored in PortletPreferences by dataset tags

Reply:
Oh, I see. Got it :)